### PR TITLE
DYN-4121: Publishing a package copies it to root directory instead of /packages so it never loads.

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:

* remove outdated logic

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### CODE REVIEW

File: src/DynamoPackages/PackageManagerExtension.cs [hunk 1]
```diff
@@ -161,18 +161,8 @@ public void Startup(StartupParams startupParams)
...
-            if (startupParams.Preferences is PreferenceSettings preferences)
-            {
-                packageUploadDirectory = string.IsNullOrEmpty(preferences.SelectedPackagePathForInstall) ? 
-                    startupParams.PathManager.DefaultPackagesDirectory : preferences.SelectedPackagePathForInstall;
-            }
-            else
-            {
-                packageUploadDirectory = startupParams.PathManager.DefaultPackagesDirectory;
-            }
+            var packageUploadDirectory = startupParams.PathManager.DefaultPackagesDirectory;
```
Comment (sm6srw) [inline src/DynamoPackages/PackageManagerExtension.cs:164]: This logic is outdated and not needed anymore. `DefaultPackagesDirectory` will gives what we want in all cases.

**Review by mjkkirschner — APPROVED** [review]
LGTM

**Comment by aparajit-pratap** [comment]
Ok, looks like I broke it. 😢 
LGTM!

**Review by aparajit-pratap — APPROVED** [review]

**Comment by QilongTang** [comment]
Merging to see if master build will be unblocked. I suspect the excel failures still require manual fixes

**Comment by QilongTang** [comment]
Even after merging I still see the following regression on recent master-15 build
```
1) Failed : Dynamo.Tests.CrashReportingTests.NoLoadedPackages
  Expected: <empty>
  But was:  < <Dynamo.PackageManager.Package> >
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at Dynamo.Tests.CrashReportingTests.NoLoadedPackages() in E:\Builds\Dynamo_master\Dynamo\test\DynamoCoreWpfTests\CrashReportingTests.cs:line 154

2) Failed : DynamoCoreWpfTests.PackageManagerUITests.PackageManagerLoadManuallyUnloadedBuiltIn
  Expected: 1
  But was:  2
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at DynamoCoreWpfTests.PackageManagerUITests.PackageManagerLoadManuallyUnloadedBuiltIn() in E:\Builds\Dynamo_master\Dynamo\test\DynamoCoreWpfTests\PackageManagerUITests.cs:line 437

3) Failed : DynamoCoreWpfTests.PackageManagerUITests.PackageManagerNoDuplicatesOnPathIsRemovedThenAdded
  Expected: 0
  But was:  1
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at DynamoCoreWpfTests.PackageManagerUITests.PackageManagerNoDuplicatesOnPathIsRemovedThenAdded() in E:\Builds\Dynamo_master\Dynamo\test\DynamoCoreWpfTests\PackageManagerUITests.cs:line 488

4) Failed : DynamoCoreWpfTests.PackageManagerUITests.PackageManagerUninstallCommand
  Expected: 1
  But was:  2
   at NUnit.Framework.Assert.That(Object actual, IResolveConstraint expression, String message, Object[] args)
   at DynamoCoreWpfTests.PackageManagerUITests.PackageManagerUninstallCommand() in E:\Builds\Dynamo_master\Dynamo\test\DynamoCoreWpfTests\PackageManagerUITests.cs:line 380
```

**Pull request merged**